conans/client/cmake.py

```diff
@@ -71,6 +72,33 @@ def _generator(self):
 
         raise ConanException("Unknown cmake generator for these settings")
 
+    def _cmake_compiler_options(self, target_os, target_arch, target_comp):
+        host_platform = platform.system()
+        same_platform = (target_os == host_platform)
+
+        cmake_flags = []
+        cmake_flags.append("-DCMAKE_SYSTEM_NAME=%s" % target_os)
+
+        if target_comp == "clang":
+            cmake_flags.append("-DCMAKE_C_COMPILER=clang")
+            cmake_flags.append("-DCMAKE_CXX_COMPILER=clang++")
+        elif target_comp == "gcc":
+            if target_os == "Windows" and not same_platform:
+                if target_arch == "x86":
+                    cmake_flags.append("-DCMAKE_C_COMPILER=i686-w64-mingw32-gcc")
+                    cmake_flags.append("-DCMAKE_CXX_COMPILER=i686-w64-mingw32-g++")
+                else:
+                    cmake_flags.append("-DCMAKE_C_COMPILER=x86_64-w64-mingw32-gcc")
+                    cmake_flags.append("-DCMAKE_CXX_COMPILER=x86_64-w64-mingw32-g++")
+            else:
+                cmake_flags.append("-DCMAKE_C_COMPILER=gcc")
```

Comment:
Yes, I am trying to use it for MinGW in Windows, and it is failing, because my $CXX variable is being dismissed, CMAKE_CXX_COMPILER prevails, which points to "g++", which is the default MinGW installation in the path, and not the one that I am pointing with my $CXX environment variable.

My concern is that I am not the only one doing this, I am aware of some other users, using MinGW and specifying the compiler via $CXX, that would have their setups and builds broken if we merge this, so we have to be very careful. I will get back to this as soon as possible, but this release iteration is becoming very busy, sorry for the delays.